R/expr-is.R

```diff
@@ -1,38 +1,68 @@
-#' Check whether a parse table corresponds to a certain expression
+#' @title Check whether a parse table corresponds to a certain expression
+#' @name pd_is
 #'
 #' @param pd A parse table.
-#' @name pd_is
+#' @param tilde_pos Integer vector indicating row-indices that should be
+#'   checked for tilde. See 'Details'.
+#'
+#' @family third-party style guide helpers
 #' @keywords internal
 NULL
 
-#' @describeIn pd_is Checks whether `pd` contains an expression wrapped in
-#'   curly brackets.
-#' @keywords internal
+#' @describeIn pd_is Checks whether `pd` contains an expression wrapped in curly brackets.
+#' @examples
+#' code <- "if (TRUE) { 1 }"
+#' pd <- compute_parse_data_nested(code)
+#' is_curly_expr(pd)
+#' is_curly_expr(pd$child$`17`$child$`14`)
+#'
+#' @export
 is_curly_expr <- function(pd) {
   if (is.null(pd)) {
     return(FALSE)
   }
   pd$token[1L] == "'{'"
 }
 
+#' @describeIn pd_is Checks whether `pd` contains a `for` loop.
+#' @examples
+#' code <- "for (i in 1:5) print(1:i)"
+#' pd <- compute_parse_data_nested(code)
+#' is_for_expr(pd)
+#' is_for_expr(pd$child$`30`)
+#'
+#' @export
 is_for_expr <- function(pd) {
   pd$token[1L] == "FOR"
 }
 
 #' @describeIn pd_is Checks whether `pd` contains is a conditional expression.
-#' @keywords internal
-is_cond_expr <- function(pd) {
+#' @examples
+#' code <- "if (TRUE) x <- 1 else x <- 0"
+#' pd <- compute_parse_data_nested(code)
+#' is_conditional_expr(pd)
+#' is_conditional_expr(pd$child$`24`)
+#'
+#' @export
+is_conditional_expr <- function(pd) {
```

Comment:
I think this is a better name than `is_cond_expr()`.

Reply:
I think we should keep it as is (or change all the `*_expr()` functions).

Reply:
But why do we need to change all the `*_expr()` functions? Their current names are expressive as they are.

Reply:
no sorry I read your comment wrongly. I thouhg you wanted `*_expression()` everywhere, but that is (obviously) not the case.

Reply:
I see.

Okay, then this should be ready for your review. I don't have any additional changes in mind.